vllm/worker/worker.py

```diff
@@ -54,10 +54,22 @@ def init_model(self):
         self.rank = self.rank if self.rank is not None else int(
             os.getenv("RANK", "-1"))
         local_rank = int(os.getenv("LOCAL_RANK", "0"))
-        self.device = torch.device(f"cuda:{local_rank}")
         if self.rank < 0:
             raise ValueError("Invalid or unspecified rank.")
-        torch.cuda.set_device(self.device)
+
+        # This caused problem for rank non-0 (for example, 1 when -tp 2), when calling torch.cuda.set_device(self.device) in ROCm.
+        # HIP Error invalid device ordial
+        # where CUDA_VISIABLE_DEVICES=0, and set_device with cuda:1.
+        try:
+            self.device = torch.device(f"cuda:{local_rank}")
+            torch.cuda.set_device(self.device)
+        except RuntimeError as re:
+            print(
+                f"RuntimeError {re} in cuda.set_device {self.device}, visible device={os.environ.get('CUDA_VISIBLE_DEVICES')}. "
```

Comment:
NIT: should this be HIP_VISIBLE_DEVICES?  even CUDA_VISIBLE_DEVICES also works.
Checking to maintain consistence across to use HIP_VISIBLE_DEVICES or CUDA_VISIBLE_DEVICES.